proposals/1/patch-steps.md

```diff
@@ -0,0 +1,139 @@
+# Patch Steps: A Standardized Extended Patch Format
+
+## Description Of Issue
+
+The current CCLoader patch format appears to be the closest for a de-facto standard for JSON patching there is.
+
+It's effective on objects, but has flaws elsewhere.
+
+It cannot handle arrays effectively, and if CrossCode game objects are being copied from other files, they end up copied verbatim.
+
+The format as-is can't be extended, but always involves a tree of objects, so a transparent replacement could be introduced without changing the formats that use patches.
+
+This would require a new format, though.
+
+## Solution
+
+A new format based on an Array root with a series of steps allows for three workflows:
+
+1. Automatic generation of a patch as before, but with better (i.e. less likely to break) handling of addition of elements to arrays and such
```

Comment:
Less likely to break as compared to what method?

Reply:
As compared to the object-based method. As noted previously it works for most stuff, but specifically not arrays.